services/account.js

```diff
@@ -1,21 +1,65 @@
+const stampit = require('@stamp/it');
+
+const {
+    createAccount,
+    deleteAccount,
+    getAccountById,
+    updateAccount,
+} = require('../db/methods/account');
+const { USE_DYNAMODB } = require('../features');
 const SequelizeAccounts = require('./sequelize/account');
 
-const AccountService = {
-    createAccount(accountId, { fundedAccountNeedsDeposit } = {}) {
-        return SequelizeAccounts.createAccount(accountId, { fundedAccountNeedsDeposit });
+const AccountService = stampit({
+    props: {
+        db: {
+            createAccount,
+            deleteAccount,
+            getAccountById,
+            updateAccount,
+        },
+        sequelize: SequelizeAccounts,
     },
+    methods: {
+        createAccount(accountId, { fundedAccountNeedsDeposit } = {}) {
+            if (!USE_DYNAMODB) {
+                return this.sequelize.createAccount(accountId, { fundedAccountNeedsDeposit });
+            }
+            return this.db.createAccount({ accountId, fundedAccountNeedsDeposit });
+        },
 
-    async deleteAccount(accountId) {
-        return SequelizeAccounts.deleteAccount(accountId);
-    },
+        async deleteAccount(accountId) {
+            if (!USE_DYNAMODB) {
+                return this.sequelize.deleteAccount(accountId);
+            }
+            return this.db.deleteAccount(accountId);
+        },
 
-    getAccount(accountId) {
-        return SequelizeAccounts.getAccount(accountId);
-    },
+        getAccount(accountId) {
+            if (!USE_DYNAMODB) {
+                return this.sequelize.getAccount(accountId);
+            }
+            return this.db.getAccountById(accountId);
+        },
+
+        async getOrCreateAccount(accountId) {
+            if (!USE_DYNAMODB) {
+                return this.createAccount(accountId);
+            }
+            const account = await this.getAccount(accountId);
+            if (account) {
+                return account;
+            }
+
+            return this.createAccount(accountId);
+        },
 
-    async setAccountRequiresDeposit(accountId, requiresDeposit) {
-        return SequelizeAccounts.setAccountRequiresDeposit(accountId, requiresDeposit);
+        async setAccountRequiresDeposit(accountId, requiresDeposit) {
```

Comment:
Is it worth creating an `updateAccount()` method to replace this to remove the need for more fine grained methods as we develop out the `account` schema?

Reply:
I think so, at least eventually. Initially I tried to create methods based on the mutations we made but partway through I abandoned that because it doesn't quite match up with our usage. If we do extend the accounts model then definitely worth exposing a generic update method at the service layer.